RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticHitToClusterAssociator.h

```diff
@@ -0,0 +1,288 @@
+                            {
+                                if(HitToRealisticSimCluster_[hitId][i] != clId && RealisticSimClusters_[HitToRealisticSimCluster_[hitId][i]].isVisible())
+                                {
+                                    float oldFraction = HitToRealisticEnergyFraction_[hitId][i];
+                                    float newFraction = oldFraction*correction;
+                                    float oldEnergy = oldFraction*totalEnergy_[hitId];
+
+                                    float newEnergy= newFraction*totalEnergy_[hitId];
+                                    float sharedEnergy = newEnergy-oldEnergy;
+                                    energyInNeighbors[HitToRealisticSimCluster_[hitId][i]] +=sharedEnergy;
+                                    totalSharedEnergy +=sharedEnergy;
+                                    RealisticSimClusters_[HitToRealisticSimCluster_[hitId][i]].increaseEnergy(sharedEnergy);
+                                    RealisticSimClusters_[HitToRealisticSimCluster_[hitId][i]].modifyFractionForHitId(newFraction, hitId);
+                                    HitToRealisticEnergyFraction_[hitId][i] = newFraction;
+                                    if(newFraction > exclusiveFraction)
+                                    {
+
+                                        RealisticSimClusters_[HitToRealisticSimCluster_[hitId][i]].increaseExclusiveEnergy(sharedEnergy);
+                                        if(oldFraction <=exclusiveFraction)
+                                        {
+                                            RealisticSimClusters_[HitToRealisticSimCluster_[hitId][i]].increaseExclusiveEnergy(oldEnergy);
+                                        }
+                                    }
+
+                                }
+                            }
+                        }
+                    }
+
+                    for(auto& elt : hAndF)
+                    {
+                        unsigned int hitId = elt.first;
+                        if(HitToRealisticSimCluster_[hitId].size()==1)
+                        {
+                            for (auto& pair: energyInNeighbors)
+                            {
+                                // hits that belonged completely to the absorbed cluster are redistributed
+                                // based on the fraction of energy shared in the shared hits
+                                float sharedFraction = pair.second/totalSharedEnergy;
+                                float assignedEnergy = totalEnergy_[hitId]*sharedFraction;
+                                RealisticSimClusters_[pair.first].increaseEnergy(assignedEnergy);
+                                RealisticSimClusters_[pair.first].addHitAndFraction(hitId, sharedFraction);
+                                if(sharedFraction > exclusiveFraction)
+                                RealisticSimClusters_[pair.first].increaseExclusiveEnergy(assignedEnergy);
```

Comment:
Please indent this line further to indicate it's within the scope of the if statement